src/main/java/org/cloudfoundry/promregator/lite/cfaccessor/SimpleCFAccessorImpl.kt

```diff
@@ -0,0 +1,511 @@
+import reactor.core.publisher.Mono
+import java.net.InetAddress
+import java.net.UnknownHostException
+import java.util.concurrent.ConcurrentHashMap
+import java.util.regex.Pattern
+import org.cloudfoundry.client.v2.PaginatedRequest as PaginatedRequestV2
+import org.cloudfoundry.client.v2.Resource as ResourceV2
+import org.cloudfoundry.client.v2.applications.ApplicationResource as ApplicationResourceV2
+import org.cloudfoundry.client.v2.applications.ListApplicationsRequest as ListApplicationsRequestV2
+import org.cloudfoundry.client.v2.applications.ListApplicationsResponse as ListApplicationsResponseV2
+import org.cloudfoundry.client.v2.organizations.ListOrganizationsRequest as ListOrganizationsRequestV2
+import org.cloudfoundry.client.v2.organizations.ListOrganizationsResponse as ListOrganizationsResponseV2
+import org.cloudfoundry.client.v2.organizations.OrganizationResource as OrganizationResourceV2
+import org.cloudfoundry.client.v2.spaces.ListSpacesRequest as ListSpacesRequestV2
+import org.cloudfoundry.client.v2.spaces.ListSpacesResponse as ListSpacesResponseV2
+import org.cloudfoundry.client.v2.spaces.SpaceResource as SpaceResourceV2
+import org.cloudfoundry.client.v3.PaginatedRequest as V3PaginatedRequest
+import org.cloudfoundry.client.v3.applications.ApplicationResource as ApplicationResourceV3
+import org.cloudfoundry.client.v3.applications.ListApplicationsRequest as ListApplicationsRequestV3
+import org.cloudfoundry.client.v3.applications.ListApplicationsResponse as ListApplicationsResponseV3
+
+private val log = mu.KotlinLogging.logger {  }
+
+@Component
+class SimpleCFAccessorImpl(
+        private val cf: CloudFoundryConfig,
+        private val promregatorConfiguration: PromregatorConfiguration,
+        private val meterRegistry: MeterRegistry,
+        ) {
+    private val rateLimiter = RateLimiter(600, meterRegistry)
+    private var _isV3Enabled = false
+    val isV3Enabled: Boolean get() = _isV3Enabled
+
+    private val cloudFoundryClients = ConcurrentHashMap<String, ReactorCloudFoundryClient>()
+
+    init {
+        meterRegistry.gauge("promregator.cf.ratelimit.available", rateLimiter) { it.availableTokens.toDouble() }
+        meterRegistry.gauge("promregator.cf.ratelimit.queue", rateLimiter) { it.waitQueue.toDouble() }
+        meterRegistry.more().counter("promregator.cf.ratelimit.wait_seconds", Tags.empty(), rateLimiter) { (it.totalWaitTimeMs / 1000.0) }
+        createClients()
+    }
+
+    private suspend fun <T> rateLimitAndRecord(requestType: RequestType, block: suspend () -> T): T? {
+        return rateLimiter.whenTokenAvailable {
+            val timer = Timer.start()
+            val result = try {
+                 block()
+            } catch (e: Exception) {
+                log.error(e) { "Error making request type: $requestType"  }
+                null
+            }
+            timer.stop(meterRegistry.timer("promregator.cffetch.single", "type", requestType.metricName))
```

Comment:
I separated out the `single` page calls as a separate timer since these added together would be double counted with the overall cffetch calls.